Sming/Makefile

```diff
@@ -554,10 +554,11 @@ dist-clean: clean samples-clean third-party-clean libraries-clean tools-clean
  	done
 	$(Q) $(GIT) checkout $(USER_LIBDIR)
 
+# Recursive wildcard search
+rwildcard = $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2) $(filter $(subst *,%,$2),$d))
```

Comment:
There is one potential issue that comes to my mind with that code: it looks for folders that start with X but are not exactly X . That means the if someone decides to have a directory called "samples.last" (a copy of the samples up to now) then the code above will find "samples" AND "samples.last" but not just "samples".

Reply:
I'm not particularly fluent with make - found that elsewhere and it looked reasonable. If I understand correctly), then creating copies of a sample, or the sample folder itself, should fail. Testing this I get expected (correct) results:

```
../samples/Basic_Interrupts/include/user_config.h
../samples/Basic_NFC/app/application.cpp
../samples/Basic_NFC/include/user_config.h
../samples/Basic_Neopixel/app/application.cpp
../samples/Basic_Neopixel/include/user_config.h
../samples/Basic_ProgMem/app/FlashData.cpp
../samples/Basic_ProgMem/app/TestProgmem.cpp
../samples/Basic_ProgMem/app/application.cpp
../samples/Basic_ProgMem/include/FlashData.h
../samples/Basic_ProgMem/include/TestProgmem.h
../samples/Basic_ProgMem/include/user_config.h
../samples/Basic_ProgMem.last/app/FlashData.cpp
../samples/Basic_ProgMem.last/app/TestProgmem.cpp
../samples/Basic_ProgMem.last/app/application.cpp
../samples/Basic_ProgMem.last/include/FlashData.h
../samples/Basic_ProgMem.last/include/TestProgmem.h
../samples/Basic_ProgMem.last/include/user_config.h
../samples/Basic_ScannerI2C/app/application.cpp
../samples/Basic_ScannerI2C/include/user_config.h
../samples/Basic_Serial/app/application.cpp
../samples/Basic_Serial/include/SerialReadingDelegateDemo.h
../samples/Basic_Serial/include/user_config.h
../samples/Basic_Servo/app/application.cpp
../samples/Basic_Servo/include/user_config.h
../samples/Basic_SmartConfig/app/application.cpp
```
...
```
../samples/Websocket_Client/app/application.cpp
../samples/Websocket_Client/include/ssl/cert.h
../samples/Websocket_Client/include/ssl/private_key.h
../samples/Websocket_Client/include/user_config.h
../samples.last/Basic_ProgMem/app/FlashData.cpp
../samples.last/Basic_ProgMem/app/TestProgmem.cpp
../samples.last/Basic_ProgMem/app/application.cpp
../samples.last/Basic_ProgMem/include/FlashData.h
../samples.last/Basic_ProgMem/include/TestProgmem.h
../samples.last/Basic_ProgMem/include/user_config.h
```

Reply:
When the task is to change the coding style of "SmingCore" and its subdirectories and "samples" and its subdirectories then changing the coding style also for "samples.last" and its subdirectories is not correct. 

It is not a big deal at the moment but it can cause us problems later due to this side-effect.

Reply:
Ah, I understand. This seems to work:

```
# Recursive wildcard search
rwildcard = $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2) $(filter $(subst *,%,$2),$d))
 
# Files that should follow our coding standards
CS_DIRS = SmingCore ../samples
CS_FILES := $(call rwildcard,$(patsubst %,%/*,$(CS_DIRS)),*.cpp *.h *.c)

cs:
	$(Q) for FILE in $(CS_FILES); do \
		echo $$FILE; \
	done

```

Reply:
Yep, seems to work also for me. Can you update the Makefile in the PR ?